lib/components/sync-status/get-note-titles.ts

```diff
@@ -1,24 +1,25 @@
-import { compact } from 'lodash';
 import noteTitleAndPreview from '../../utils/note-utils';
 
-const getNoteTitles = (ids, notes, limit = Infinity) => {
-  const matchedNotes = ids.map((id, i) => {
-    if (i >= limit) {
-      return;
-    }
+import * as T from '../../types';
 
-    const note = notes.find(thisNote => thisNote.id === id);
-
-    if (!note) {
-      // eslint-disable-next-line no-console
-      console.log(`Could not find note with id '${id}'`);
-      return null;
-    }
+type NoteTitle = {
+  id: String;
+  title: String;
+};
 
-    return { id, title: noteTitleAndPreview(note).title };
-  });
+const getNoteTitles = (
+  ids: String[] = [],
```

Comment:
it's safe to use `T.EntityId` here since we're indexing the `notes` with it. `String` works because `EntityId` is an alias for `string` but we might as well leave that more specific.

Also note that in a confusing way `string` and `String` are different types. `String` should generally be reserved for places where we create `new String("something")` and the same goes for the other capitalized types.

Reply:
Duh, I shoulda know that.

Also, was planning to inquire about `string` and `String`

Reply:
Fixed in fffcc43